src/PetProject.Framework.Kafka/Consumer/Consumer.cs

```diff
@@ -60,100 +72,174 @@ public bool StartConsuming()
+            {
+                var key = new StringDeserializer(Encoding.UTF8).Deserialize(message.Topic, message.Key);
+                var messageContent = new JsonDeserializer<MessageWrapper>().Deserialize(message.Topic, message.Value);
+
+                var report = producer.ProduceAsync(message.Topic, key, messageContent).Result;
+
+                if (!report.Error.HasError)
+                {
+                    return;
+                }
+
+                this.logger.KafkaLogCritical("Critical Error when retrying to send message. Topic: {topic} | Message: {message} | Timestamp: {timestamp} | Error: {error}", message.Topic, messageContent, report.Timestamp, report.Error);
+            }
+        }
 
         /// <summary>
-        /// Must be implemented by the client in order to consumer messages and add custom treatment.
+        /// May be overriden by the client in order to consumer messages and add custom treatment.
         /// </summary>
         /// <param name="sender">Message sender.</param>
-        /// <param name="message">Message envelope with the content to consume.</param>
-        protected void HandleMessage(object sender, Message<string, string> consumerMessage)
+        /// <param name="consumerMessage">Message envelope with key value pair content.</param>
+        protected void HandleMessage(object sender, Message<string, MessageWrapper> consumerMessage)
         {
             if (consumerMessage.Value == null)
             {
+                this.logger.KafkaLogInfo("ConsumerMessage has no content: {consumerMessage}", consumerMessage);
                 return;
             }
 
-            MessageWrapper wrappedMessage;
-            try
-            {
-                wrappedMessage = JsonConvert.DeserializeObject<MessageWrapper>(consumerMessage.Value, this.settings);
-            }
-            catch (Exception)
-            {
-                throw new Exception("Consumer error when deserializing.");
-            }
+            var wrappedMessage = consumerMessage.Value;
+
+            this.CallHandler(wrappedMessage);
+        }
 
+        protected void CallHandler(MessageWrapper wrappedMessage)
+        {
             var type = Type.GetType(wrappedMessage.MessageType);
 
             if (!this.messageHandlers.ContainsKey(type))
             {
-                throw new Exception("An handler for this type of message does not exist. Please define one with ConsumerHandlerFor<> method!");
+                this.logger.KafkaLogError("An handler for this {type} of message does not exist. Please define one with Receive<> method!", type);
+                return;
             }
 
             this.messageHandlers[type].DynamicInvoke(wrappedMessage.Message);
         }
+
+        private void StopConsumer()
```

Comment:
I think we are mixing "stop consumer" logic with "dispose" logic. Dispose should call stop if the consumer is running but stop shouldn't invoke "dispose" logic

Reply:
Yep will fix it, Forgot to remove after testing.

Reply:
Also bad naming it shouldn't be called stopconsumer